src/main/java/service/SupplyOperationHandler.java

```diff
@@ -0,0 +1,13 @@
+package service;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class SupplyOperationHandler implements OperationHandler {
+
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
```

Comment:
Consider validating the `transaction` and `storage` parameters to ensure they are not `null`. This can prevent `NullPointerException` and ensure that your method is robust.